pagopa/io-app
# feat(Firma con IO): [SFEQS-1060] Add qtsp clauses and document preview screens

## Short description
In this PR, a screen has been added to show the QTSP clauses that the user must accept. In a clause there should be one or more links to a pdf document. Some clause other than the regular URL may contain a TAG_LINK.

```
{
  clauses: [
    {
      text: This is the clause text [text to link](https://normalLink/example.pdf) with more links [text to link](@TAG_LINK)
    }
  ]
}
```

You can map TAG_LINKS to replace them with other URLs. TAG_LINKS was introduced because in the FCI signature flow some clauses have a link to be replaced dynamically with other contained user data. (This could be a temporarily solutions because changes could be by QTSP)

https://user-images.githubusercontent.com/49342144/205942495-684d9397-4edc-482d-a7c3-6857d4ab89fc.mov

## List of changes proposed in this pull request
- Added `FciQtspClausesScreen`
- Added `FciDocumentPreviewScreen`
- Added `DocumentViewer` component
- Added `QtspSignatureField` and `LinkedText` components
- Added a saga to handle the download of pdf document

## How to test
* if not yet merged get this branch https://github.com/pagopa/io-dev-api-server/tree/SFEQS-1082-fci-endpoints from io-dev-server-api
* enable FCI_ENABLED ff
* follow signature flow
* on the QTSP clauses screen try to tap on a link to show related pdf in preview
* the continue button it should be activate when all checkbox are selected
